lib/container.ex

```diff
@@ -3,29 +3,51 @@ defmodule FileStorageApi.Container do
   Module for handling asset containers
   """
 
+  import FileStorageApi.Base
+
+  alias FileStorageApi.API.Azure.Container, as: AzureContainer
+  alias FileStorageApi.API.S3.Container, as: S3Container
+
   @type t :: %__MODULE__{
           name: String.t(),
           files: [FileStorageApi.File.t()],
           max_results: non_neg_integer,
           next_marker: String.t(),
           date: DateTime.t()
         }
-  @callback create(String.t(), map) :: {:ok, map} | {:error, map}
+  @callback create(String.t(), atom, map) :: {:ok, map} | {:error, map}
   @type options :: [{:max_results, non_neg_integer} | {:marker, String.t()}]
-  @callback list_files(String.t(), options) :: {:ok, [__MODULE__.t()]} | {:error, map}
+  @callback list_files(String.t(), atom, options) :: {:ok, [__MODULE__.t()]} | {:error, map}
 
   defstruct name: nil, files: [], max_results: nil, next_marker: nil, date: nil
 
-  import FileStorageApi.Base
-
   @doc """
   Will create container with binary as input for bucket name
 
   Opts allows for setting cors_policy as map or true will only be applied to S3
   """
   @spec create(String.t(), map) :: any
   def create(container_name, opts \\ %{}) do
-    api_module(Container).create(container_name, opts)
+    connection_name =
+      if Map.has_key?(opts, :container_name) do
+        opts[:container_name]
+      else
+        :default
+      end
+
+    module_container =
```

Comment:
This should really be more dynamic like before not sure why you remove api_module and replace it with this adds so much more code and complexity everywhere.
This all should be handled in the base module.

Reply:
Something like api_module(Container, connection_name) should return the correct module for use

Reply:
I couldn't search through this project.

function `api_module` return "something", and it would invoke `.upload` on that "something". It took me quite a while to figure out which module was being called at what point.
Since the code is the only exact documentation we have, the code should help me navigate. And `api_module` did not help.

Disclaimer:
After this refactor is became clear that to code is quite nicely set up. But that is only noticeable after understanding the code. So that is why I didn't put it back: it makes the code way too hard to read

Reply:
I thought about letting the API-module construct the module-name instead of doing it here. But the module-name should really be known in this module, so it is clear what parts of the code-base you're referring to.
That's why I kept it like this